exporter/prometheus/prometheus_test.go

```diff
@@ -342,3 +342,88 @@ func TestCumulativenessFromHistograms(t *testing.T) {
 		t.Fatalf("\ngot:\n%s\n\nwant:\n%s\n", got, want)
 	}
 }
+
+func TestConstLabelsIncluded(t *testing.T) {
+	constLabels := prometheus.Labels{
+		"service": "spanner",
+	}
+	measureLabel, _ := tag.NewKey("method")
+
+	exporter, err := NewExporter(Options{
+		ConstLabels: constLabels,
+	})
+	if err != nil {
+		t.Fatalf("failed to create prometheus exporter: %v", err)
+	}
+	view.RegisterExporter(exporter)
+	defer view.UnregisterExporter(exporter)
+
+	names := []string{"foo", "bar", "baz"}
+
+	var measures mSlice
+	for _, name := range names {
+		measures.createAndAppend("tests/"+name, name, "")
+	}
+
+	var vc vCreator
+	for _, m := range measures {
+		vc.createAndAppend(m.Name(), m.Description(), []tag.Key{measureLabel}, m, view.Count())
+	}
+
+	if err := view.Register(vc...); err != nil {
+		t.Fatalf("failed to create views: %v", err)
+	}
+	defer view.Unregister(vc...)
+
+	view.SetReportingPeriod(time.Millisecond)
+
+	ctx, _ := tag.New(context.Background(), tag.Upsert(measureLabel, "issue961"))
+	for _, m := range measures {
+		stats.Record(ctx, m.M(1))
+	}
+
+	srv := httptest.NewServer(exporter)
+	defer srv.Close()
+
+	var i int
+	var output string
+	for {
+		time.Sleep(10 * time.Millisecond)
+		if i == 1000 {
+			t.Fatal("no output at /metrics (10s wait)")
+		}
+		i++
+
+		resp, err := http.Get(srv.URL)
+		if err != nil {
+			t.Fatalf("failed to get /metrics: %v", err)
+		}
+
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatalf("failed to read body: %v", err)
+		}
+		resp.Body.Close()
+
+		output = string(body)
+		if output != "" {
+			break
+		}
+	}
+
+	if strings.Contains(output, "collected before with the same name and label values") {
+		t.Fatal("metric name and labels being duplicated but must be unique")
+	}
+
+	if strings.Contains(output, "error(s) occurred") {
+		t.Fatal("error reported by prometheus registry")
+	}
+
+	for _, name := range names {
```

Comment:
I actually meant literally 
want := `tests_foo{method="issue961",service="spanner"} 1
tests_bar{method="issue961",service="spanner"} 1`
instead of this iterations on the names
and then wholly compare the outputs i.e.
```go
if output != want {
  t.Fatal
}
```

Reply:
Ah, I see!

Reply:
I've removed the iteration, hopefully this is now what you were expecting :).